component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/SpaceDAOImpl.java

```diff
@@ -88,4 +88,50 @@ public SpaceEntity getSpaceByPrettyName(String spacePrettyName) {
     }
   }
 
+  @Override
+  public List<SpaceEntity> getCommonSpaces(String userId, String profileUserId, int offset, int limit) {
+    if (userId == null || userId == "") {
+      throw new IllegalArgumentException("userId is null or equals to 0");
+    }
+    if (profileUserId == null || profileUserId == "") {
+      throw new IllegalArgumentException("profileUserId is null or equals to 0");
+    }
+    if (offset < 0) {
+      throw new IllegalArgumentException("offset must be positive");
+    }
+    if (limit <= 0) {
+      throw new IllegalArgumentException("limit must be > 0");
+    }
+    TypedQuery<SpaceEntity> query = getEntityManager().createNamedQuery("SpaceEntity.getCommonSpacesBetweenTwoUsers",
+            SpaceEntity.class);
+    query.setParameter("user1", userId);
+    query.setParameter("user2", profileUserId);
+    query.setFirstResult(offset);
+    query.setMaxResults(limit);
+    return query.getResultList();
+
+  }
+
+  @Override
+  public int countCommonSpaces(String userId, String profileUserId) {
+    if (userId == null || userId == "") {
+      throw new IllegalArgumentException("userId is null or equals to 0");
+    }
+    if (profileUserId == null || profileUserId == "") {
+      throw new IllegalArgumentException("profileUserId is null or equals to 0");
+    }
+
+    Query q = getEntityManager().createNativeQuery("SELECT count(*) FROM SOC_SPACES t where t.SPACE_ID in " +
```

Comment:
Can you add it as NamedQuery

Reply:
in fact, it throws UndeclaredThrowableException because HQL did not support join queries so i had to use SQL query type,